sensor/kubernetes/listener/resources/store_provider.go

```diff
@@ -44,7 +44,7 @@ func InitializeStore() *StoreProvider {
 	podStore := newPodStore()
 	svcStore := newServiceStore()
 	nodeStore := newNodeStore()
-	entityStore := clusterentities.NewStoreWithMemory(uint16(memSizeSetting), debugClusterEntitiesStore.BooleanSetting())
+	entityStore := clusterentities.NewStore()
```

Comment:
nitpick: I would keep an option to set `debugMode`.

Maybe to set `memSizeSetting` to `0` before:
```
	log.Infof("Initializing cluster entities store with memory that will last for %d ticks", memSizeSetting)
```
In that case, logs will not be misleading. We should also add a comment in the code with the ticket number before that.

```
	// TODO(ROX-28242): Disable ROX_PAST_CLUSTER_ENTITIES_MEMORY_SIZE option.
	memSizeSetting = 0
	log.Infof("Initializing cluster entities store with memory that will last for %d ticks", memSizeSetting)
```

I'm not sure if some documentation/changelog changes are required.

Reply:
Good point. I've hardcoded the `memSizeSetting` to zero like you suggested.